src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepDynamicContextTest.java

```diff
@@ -207,4 +215,31 @@ public class ExecutorStepDynamicContextTest {
         });
     }
 
+    @Test public void onceRetentionStrategyNodeDisappearance() throws Throwable {
+        logging.recordPackage(ExecutorStepExecution.class, Level.FINE).record(FlowExecutionList.class, Level.FINE);
+        sessions.then(j -> {
+            // Start up a build that needs executor and then reboot and take the node offline
+            // Starting job first ensures we don't immediately fail if Node comes from a Cloud
```

Comment:
So should the test actually start the build before creating the agent, and wait for it to wait for an executor?

Reply:
Doesn't matter actually.